pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -751,14 +757,19 @@ public static Map<String, List<KitRequestShipping>> getAllKitRequestsByRealm(@No
     }
 
     private static PreparedStatement getPreparedStatement(@NonNull Connection conn, String target, @NonNull String realm, String type,
-                                                          boolean getAll) throws SQLException {
+                                                          Integer kitTypeId, boolean getAll) throws SQLException {
         PreparedStatement stmt = null;
         if (getAll) {
             String query = SQL_SELECT_KIT_REQUEST.concat(QueryExtension.BY_REALM);
             stmt = conn.prepareStatement(query);
             stmt.setString(1, realm);
         } else {
-            if (StringUtils.isNotBlank(realm) && StringUtils.isNotBlank(type)) {
+            if (kitTypeId != null && kitTypeId != null) {
+                    String query = addQueryExtension(target, SQL_SELECT_KIT_REQUEST.concat(QueryExtension.BY_REALM_AND_TYPE_ID));
+                    stmt = conn.prepareStatement(query);
+                    stmt.setString(1, realm);
+                    stmt.setInt(2, kitTypeId);
```

Comment:
Not sure I understand, looking at the next clause... `kitTypeId` and `type` are interchangeable?

Reply:
Yes, it's based on what is chosen as the query extension. if it's `BY_REALM_AND_TYPE_ID` then the second one should be the type id, and if it's `BY_REALM_AND_TYPE` then the second one should be the type name. 

It would be far above this ticket's scope but we can make one ticket to refactor all of these kit methods.